Plugins/Mute/MuteManager.cs

```diff
@@ -1,21 +1,173 @@
 ﻿using SharedLibraryCore;
+using SharedLibraryCore.Database.Models;
+using SharedLibraryCore.Interfaces;
+using static System.Enum;
 
 namespace Mute;
 
 public class MuteManager
 {
-    public async Task<bool> Mute(GameEvent gameEvent)
+    private readonly IMetaServiceV2 _metaService;
+    private readonly ITranslationLookup _translationLookup;
+    private List<MuteStateMeta> _muteList = null!;
+
+    public MuteManager(IMetaServiceV2 metaService, ITranslationLookup translationLookup)
+    {
+        _metaService = metaService;
+        _translationLookup = translationLookup;
+    }
+
+    public async Task InitialiseMuteList() => _muteList = await ReadMuteList();
+    public bool HasMutes => _muteList.Any(x => x.Expiration > DateTime.UtcNow || x.Expiration == null);
+
+    public IEnumerable<MuteStateMeta> Mutes =>
+        _muteList.Where(x => x.Expiration > DateTime.UtcNow || x.Expiration == null);
+
+    public async Task<MuteStateMeta> GetCurrentMuteState(EFClient client)
+    {
+        var clientMuteMeta = await ReadPersistentDataV2(client);
+        if (clientMuteMeta is not null) return clientMuteMeta;
+
+        // Return null if the client doesn't have old or new meta.
+        var muteState = await ReadPersistentDataV1(client);
+        clientMuteMeta = new MuteStateMeta
+        {
+            ClientId = client.ClientId,
+            CleanedName = client.CleanedName,
```

Comment:
Is there a reason we're not retaining their color codes here and other locations? Webfront supports color codes if enabled, and the game of course will also.

Reply:
Oh also, how is this name being stored? If it's being stored at the time of mute and never updated that's problematic because they could change their name and it wouldn't make sense. However because this is a temporary measure until it's an official penalty type, I suppose this is fine.